storage2/src/snapshot.rs

```diff
@@ -0,0 +1,29 @@
+/// Snapshots maintain a point-in-time view of the underlying storage, suitable
+/// for read-only access by multiple threads, i.e. RPC calls.
+///
+/// This is implemented as a wrapper around a [RocksDB snapshot](https://github.com/facebook/rocksdb/wiki/Snapshot)
+/// with an associated JMT version number for the snapshot.
+pub(crate) struct Snapshot {
+    rocksdb_snapshot: rocksdb::Snapshot<'static>,
```

Comment:
Could we add a `TODO` comment highlighting that the `'static` is a temporary hack, and that we need to find a workaround as a separate issue?